### ISSUE TITLE
fix(grid): 2/3 dealiasing mask keeps the aliasing mode |k| = N/3 whenever N is divisible by 3

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#88]
Part of #80.

## Summary

`FourierGrid1D.k_dealias`, `FourierGrid1D/2D/3D.dealias_filter` compute the cutoff as `k_max * 2/3` with `k_max = N/2` and keep modes with `|k| <= cutoff`. For even `N` divisible by 3 (30, 48, 96, 192, 384, …) the mode `K = N/3` is kept. The product of two such modes lands at `2N/3`, which aliases to `−N/3`, a kept mode, at full amplitude. Orszag's 2/3 rule requires `3K < N` strictly. Every grid size used in the test suite is a power of two, so the suite cannot see it.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp
from spectraldiffx import FourierGrid1D

for N in (30, 32, 48, 96):
    g = FourierGrid1D.from_N_L(N=N, L=2 * jnp.pi, dealias="2/3")
    K = N // 3
    print(f"N={N} K=N/3={K} kept={bool(g.dealias_filter()[K])}  alias-free (3K<N)? {3*K < N}")
```

```
N=30 K=10 kept=True  alias-free (3K<N)? False
N=32 K=10 kept=True  alias-free (3K<N)? True
N=48 K=16 kept=True  alias-free (3K<N)? False
N=96 K=32 kept=True  alias-free (3K<N)? False
```

Amplitude of the spurious mode after `apply_dealias(u * u)` with `u = cos(Kx)` (exact product coefficient at `±2K` is 0.25):

```
N=30 K=10 -> 0.250   N=32 K=10 -> 0.000   N=48 K=16 -> 0.250
N=64 K=21 -> 0.000   N=96 K=32 -> 0.250   N=192 K=64 -> 0.250   N=256 K=85 -> 0.000
```

## Root cause

- `spectraldiffx/_src/fourier/grid.py:209-214` (`k_dealias`): `cutoff = k_max * 2 / 3; k = where(|k| > cutoff, 0, k)`.
- `grid.py:225-229` (`dealias_filter` 1D): `where(|k| <= cutoff, 1, 0)`.
- `grid.py:439-449` (2D) and `grid.py:741-755` (3D): same construction per axis.

Odd `N` already has a strict cutoff (`(N−1)/3`), so only even `N` divisible by 3 changes.

## Proposed fix

Work in integer mode index rather than physical wavenumber:

```python
n = jnp.fft.fftfreq(N, d=1.0 / N)          # integer mode numbers
mask = jnp.abs(n) < N / 3                   # strict: keeps |n| <= (N-1)//3
```

and use the same mask in `k_dealias` (`jnp.where(mask, k, 0.0)`) and in the 2D/3D `dealias_filter` as an outer product of per-axis 1D masks (this also removes the meshgrid rebuild noted in #82).

## Acceptance criteria

- [ ] Parametrised test over `N ∈ {30, 32, 48, 63, 64, 96}`: for `K = largest kept mode`, `apply_dealias(cos(Kx)²)` has zero energy at the alias index (< 1e-14).
- [ ] 2D/3D: same on a non-square grid with `Nx = 48, Ny = 32`.
- [ ] Existing test `test_fourier_grid_1d_k_dealias_zeros_above_cutoff` rewritten to assert alias-freeness rather than restating the implementation.
- [ ] Docstring at `grid.py:200-207` states the strict inequality.

## Notes

The `dealias` field's `Literal["2/3", "padding", None]` accepts `"padding"` (and any other string) as a silent no-op; that is tracked in the "small guards and docstrings" issue in this epic.
